src/Mono.Android/Android.Runtime/InputStreamInvoker.cs

```diff
@@ -92,7 +98,21 @@ public override int Read (byte[] buffer, int offset, int count)
 
 		public override long Seek (long offset, SeekOrigin origin)
 		{
-			throw new NotSupportedException ();
+			if (BaseFileChannel == null)
+				throw new NotSupportedException ();
+
+			switch (origin) {
+			case SeekOrigin.Begin:
+				BaseFileChannel.Position (offset);
+				break;
+			case SeekOrigin.Current:
+				BaseFileChannel.Position (BaseFileChannel.Position() + offset);
+				break;
+			case SeekOrigin.End:
+				BaseFileChannel.Position (BaseFileChannel.Size() + offset);
```

Comment:
Shouldn't this be `BaseFileChannel.Size () - offset`?  (Note **`-`** subtraction.)

Reply:
If you have a look at https://docs.microsoft.com/en-us/dotnet/api/system.io.seekorigin?view=netframework-4.8 you will see that a negative offset parameter is used. Other reference implementations: https://github.com/dotnet/aspnetcore/blob/58a107b27e7a73cf24f68b7ad38818a59b5289b9/src/Http/Http/src/Internal/ReferenceReadStream.cs#L88-L91 and https://github.com/dotnet/aspnetcore/blob/86728ecc17c339f699d21c8abd5ee93037b51a1e/src/Http/WebUtilities/src/MultipartReaderStream.cs#L113-L116

Reply:
Ah, silly me.  Callers would provide a negative `offset` value with `SeekOrigin.End`.